upup/pkg/fi/cloudup/scalewaytasks/instance.go

```diff
@@ -177,6 +180,64 @@ func (_ *Instance) RenderScw(c *fi.CloudupContext, actual, expected, changes *In
 		if err != nil {
 			return fmt.Errorf("error waiting for instance %s of group %q: %w", srv.Server.ID, fi.ValueOf(expected.Name), err)
 		}
+
+		// We update the server's infos (to get its IP)
+		server, err := instanceService.GetServer(&instance.GetServerRequest{
+			Zone:     zone,
+			ServerID: srv.Server.ID,
+		})
+		if err != nil {
+			return fmt.Errorf("error getting server %s: %s", srv.Server.ID, err)
+		}
+
+		// If instance has role master, we add its private IP to the list to add it to the lb's backend
+		for _, tag := range expected.Tags {
+			if tag == scaleway.TagNameRolePrefix+"="+scaleway.TagRoleMaster {
+				mastersPrivateIPs = append(mastersPrivateIPs, *server.Server.PrivateIP)
+			}
+		}
+	}
+
+	// If IG is master, we add the new servers' IPs to the load-balancer's back-end
+	if len(mastersPrivateIPs) > 0 {
+		lbService := cloud.LBService()
+		region := scw.Region(os.Getenv("SCW_DEFAULT_REGION"))
+
+		lbs, err := cloud.GetClusterLoadBalancers(cloud.ClusterName(expected.Tags))
+		if err != nil {
+			return fmt.Errorf("error listing load-balancers for instance creation: %w", err)
+		}
+
+		for _, loadBalancer := range lbs {
+			backEnds, err := lbService.ListBackends(&lb.ListBackendsRequest{
+				Region: region,
+				LBID:   loadBalancer.ID,
+			})
+			if err != nil {
+				return fmt.Errorf("error listing load-balancer's back-ends for instance creation: %w", err)
+			}
+			if backEnds.TotalCount > 1 {
+				return fmt.Errorf("found multiple back-ends for load-balancer %s, exiting now", loadBalancer.ID)
```

Comment:
It's not exiting.

```suggestion
				return fmt.Errorf("found multiple back-ends for load-balancer %s", loadBalancer.ID)
```

or perhaps "cannot have multiple back-ends for..."